pydsa/dfs.py

```diff
@@ -0,0 +1,17 @@
+# Depth First Search
+# Complexity: O(V+E)
+# uses stack
+
+
+def dfs(graph, start):
+    stack = list()
+    visited = list()
+    stack.append(start)
+    visited.append(start)
+    while len(stack) > 0:
+        node = stack.pop()
+        for subadjnode in graph[node]:
```

Comment:
Hi harshul

Maybe you could try something like

def dfs(graph, start, visited=None):
    if visited is None:
        visited = set()
    visited.add(start)
    for next in graph[start] - visited:
        dfs(graph, next, visited)
    return visited

It would retain the complexity O(V+E) and is the standard recursion procedure for DFS